omod/src/test/java/org/openmrs/module/webservices/rest/doc/SwaggerSpecificationCreatorTest.java

```diff
@@ -34,79 +31,110 @@
 			String tableName = rs.getString(3);
-			
+
 			ret.put(tableName, dbcon.getRowCount(tableName));
 		}
-		
+
 		return ret;
 	}
-	
+
 	@Before
 	public void init() throws Exception {
 		// init REST
 		Context.getService(RestService.class).initialize();
-		
+
 		Context.getAdministrationService().saveGlobalProperty(
 		    new GlobalProperty(RestConstants.SWAGGER_QUIET_DOCS_GLOBAL_PROPERTY_NAME, "true"));
-		
+
 		// ensure GP is written to database before we count the rows
 		Context.flushSession();
-		
+
 		beforeCounts = getRowCounts();
 	}
-	
+
 	@Test
 	public void checkNoDatabaseChanges() throws Exception {
 		SwaggerSpecificationCreator ssc = new SwaggerSpecificationCreator("/v1/");
 		ssc.BuildJSON();
-		
+
 		Map<String, Integer> afterCounts = getRowCounts();
-		
+
 		Assert.assertEquals("Ensure no tables are created or destroyed", beforeCounts.size(), afterCounts.size());
 		Assert.assertTrue("Ensure that no data was added or removed from any tables",
 		    ensureCountsEqual(beforeCounts, afterCounts));
 	}
-	
+
 	private boolean ensureCountsEqual(Map<String, Integer> beforeCounts, Map<String, Integer> afterCounts) throws Exception {
 		for (String key : beforeCounts.keySet()) {
 			if (beforeCounts.get(key) != afterCounts.get(key)) {
 				System.err.println("The " + key + " table has a different number of rows (" + beforeCounts.get(key)
 				        + " before, " + afterCounts.get(key) + " after).");
-				
+
 				return false;
 			}
 		}
-		
+
 		return true;
 	}
-	
+
 	// makes sure that every operation has a unique operationId
 	@Test
 	public void checkOperationIdsSet() throws Exception {
 		List<String> operationIds = new ArrayList<String>();
-		
+
 		SwaggerSpecificationCreator ssc = new SwaggerSpecificationCreator("/v1/");
 		ssc.BuildJSON();
 		SwaggerSpecification spec = ssc.getSwaggerSpecification();
-		
+
 		for (Path p : spec.getPaths().getPaths().values()) {
 			for (Operation o : p.getOperations().values()) {
 				Assert.assertFalse("Ensure each operation has a unique ID", operationIds.contains(o.getOperationId()));
 				operationIds.add(o.getOperationId());
 			}
 		}
 	}
+
+	// makes sure that every GET operation has the "v" parameter
+	@Test
+	public void checkRepresentationParamExists() throws Exception {
+		List<String> operationIds = new ArrayList<String>();
+
+		SwaggerSpecificationCreator ssc = new SwaggerSpecificationCreator("/v1/");
+		ssc.BuildJSON();
+		SwaggerSpecification spec = ssc.getSwaggerSpecification();
+
+		for (Path p : spec.getPaths().getPaths().values()) {
+			for (Operation o : p.getOperations().values()) {
+				if (o.getName().equals("get")) {
+					Assert.assertTrue("Ensure each GET operation has the 'v' query parameter",
+					    operationHasRepresentationParam(o));
+				}
+			}
+		}
+	}
+
+	private boolean operationHasRepresentationParam(Operation o) {
+		boolean ret = false;
+
+		for (Parameter p : o.getParameters()) {
+			if (p.getName().equals("v")) {
+				ret = true;
```

Comment:
Shouldn't we return immediately to save some unnecessary cycles? Or at least break out of the for loop.